modules/core/src/test/java/org/apache/ignite/internal/metric/JmxMetricExporterSpiTest.java

```diff
@@ -144,4 +210,125 @@ public void testFilterAndExport() throws Exception {
 
         assertEquals(44L, bean2.getAttribute("test3"));
     }
+
+    /** */
+    @Test
+    public void testCachesList() throws Exception {
+        Set<String> cacheNames = new HashSet<>(Arrays.asList("cache-1", "cache-2"));
+
+        for (String name : cacheNames)
+            ignite.createCache(name);
+
+        TabularDataSupport data = monitoringList("caches");
+
+        assertEquals(3, data.size());
+
+        for(int i=0; i<data.size(); i++) {
+            CompositeData row = data.get(new Object[] {i});
+
+            cacheNames.remove(row.get("cacheName"));
+        }
+
+        assertTrue(cacheNames.toString(), cacheNames.isEmpty());
+    }
+
+    /** */
+    @Test
+    public void testCacheGroupsList() throws Exception {
+        Set<String> grpNames = new HashSet<>(Arrays.asList("grp-1", "grp-2"));
+
+        for (String grpName : grpNames)
+            ignite.createCache(new CacheConfiguration<>("cache-" + grpName).setGroupName(grpName));
+
+        TabularDataSupport grps = monitoringList("cacheGroups");
+
+        assertEquals("ignite-sys, grp-1, grp-2", 3, grps.size());
```

Comment:
Strange error message. "3" can be replaced with `ignite.context().cache().cacheGroupDescriptors().size()`